src/client.cpp

```diff
@@ -35,6 +36,8 @@ namespace discpp {
         this->my_instance_id = next_instance_id;
         next_instance_id++;
 
+        this->command_handler = std::make_shared<CommandHandler>(*this);
+        
         client_instances.emplace(my_instance_id, this);
```

Comment:
Consider switching the ClientInstance list to shared_ptr list and use shared_from_this() here (it will probably not work out of the box as there should be at least one shared_ptr that owns *this but you can "circumvent" that by creating a shared_ptr with an empty deleter - although that may not be safe)
What you could do too is have a function that creates a bot and make users use that function instead of the constructor.